source/_docs/mqtt/discovery.markdown

```diff
@@ -222,16 +222,16 @@ The following software has built-in support for MQTT discovery:
 
 ### {% linkable_title Examples %}
 
-A motion detection device which can be represented by a [binary sensor](/components/binary_sensor.mqtt/) for your garden would sent its configuration as JSON payload to the Configuration topic. After the first message to `config`, then the MQTT messages sent to the state topic will update the state in Home Assistant.
+A motion detection device which can be represented by a [binary sensor](/components/binary_sensor.mqtt/) for your garden would send its configuration as JSON payload to the Configuration topic. After the first message to `config`, then the MQTT messages sent to the state topic will update the state in Home Assistant.
 
 - Configuration topic: `homeassistant/binary_sensor/garden/config`
 - State topic: `homeassistant/binary_sensor/garden/state`
-- Payload:  `{"name": "garden", "device_class": "motion"}`
+- Payload:  `{"name": "garden", "device_class": "motion", "state_topic": "homeassistant/binary_sensor/garden/state"}`
```

Comment:
Binary sensor still need it -> https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mqtt/discovery.py#L60

Reply:
Binary sensor needs a `state_topic`, it's just that the state topic is implicitly set if absent.
A warning is printed when this happens:
https://github.com/home-assistant/home-assistant/blob/603e2cd961e1ca7c3aae46f14e862419b223b564/homeassistant/components/mqtt/discovery.py#L294